r1cs/gadgets/crypto/src/signature/schnorr/mod.rs

```diff
@@ -76,11 +76,13 @@ for SchnorrRandomizePkGadget<G, ConstraintF, GG>
             .iter()
             .flat_map(|b| b.into_bits_le())
             .collect::<Vec<_>>();
-        let rand_pk = base.mul_bits(
-            &mut cs.ns(|| "Compute Randomizer"),
-            &public_key.pub_key,
-            randomness.iter(),
-        )?;
+        let rand_pk = {
+            let base_pow_rand = base.mul_bits(
+                &mut cs.ns(|| "Compute randomizer"),
+                randomness.iter(),
+            )?;
```

Comment:
Why not using fixed base? Do we really want to keep the signature scheme parameters as variable?

Reply:
This is a derived implementation. I think they wanted to be generic with respect to the choice of the generator